### PR TITLE
Prevent users from deselecting all transport products

### LINKED ISSUE #206: Prevent a user from deselecting all transport products

### PR DESCRIPTION
**Pull request by kas70**
closes #206.

### CODE REVIEW

File: src/de/grobox/liberario/fragments/ProductDialogFragment.java [hunk 1]
```diff
@@ -183,6 +186,13 @@ public void bindView(final ViewHolder ui) {
 				public void onClick(View v) {
 					int position = adapter.getAdapterPosition(ProductItem.this);
 					adapter.toggleSelection(position);
+					EnumSet<Product> products = getProductsFromItems(adapter.getSelectedItems());
```
Comment (grote) [inline src/de/grobox/liberario/fragments/ProductDialogFragment.java]: I think you don't need to call `getProductsFromItems()` here, just `adapter.getSelectedItems()` should be sufficient.

File: src/de/grobox/liberario/fragments/ProductDialogFragment.java [hunk 2]
```diff
@@ -183,6 +186,13 @@ public void bindView(final ViewHolder ui) {
...
+					// if no products are selected, disable the ok-button
+					if(products.size()==0){
+						fragment_okButton.setEnabled(false);
+					}else{
```
Comment (grote) [inline src/de/grobox/liberario/fragments/ProductDialogFragment.java]: coding style nitpick: space before `else`

File: sublibs/public-transport-enabler [hunk 3]
```diff
@@ -1 +1 @@
-Subproject commit 7821c7458d4a66e3e085f2f94c704b99ef3ab76f
```
Comment (grote) [inline sublibs/public-transport-enabler]: This is unrelated to your PR, please don't change it.

**Comment by grote** [comment]
Thanks for another contribution! :)

This looks good. There's just strange merge commits included. Do you know how to rebase with git? The best approach would be to work in a feature branch not in `master` and then always rebase on `origin/master`.

**Comment by kas70** [comment]
Woah, sorry. I'm having some problems with git 🙈

File: src/de/grobox/liberario/fragments/ProductDialogFragment.java (hunk 1 above)
Comment (kas70) [inline src/de/grobox/liberario/fragments/ProductDialogFragment.java]: Doesn't seem to work without ```getProductsFromItems()```.

**Comment by kas70** [comment]
I created a new fork and will open a new pull request, soon.

File: src/de/grobox/liberario/fragments/ProductDialogFragment.java (hunk 1 above)
Comment (grote) [inline src/de/grobox/liberario/fragments/ProductDialogFragment.java]: What doesn't work? You would of course not use `EnumSet<Product>` but whatever else the adapter returns. You don't care about the type, just about how many it is, right?

**Comment by kas70** [comment]
Closed in favour of https://github.com/grote/Transportr/pull/210.
